### PR TITLE
1.0.4

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
Introduced a new database for the moderation commands. Kept the mute one. Changed the audit command where now you can get all the users ban, kick and warns with their user id or a case id. Added the ability to set a welcome channel and also the ability to set a leave and welcome message. Added the ability to set a log channel. Also added getting started command and fixed the purge command.

### CODE REVIEW

File: application/build.gradle [hunk 1]
```diff
@@ -45,15 +45,13 @@ dependencies {
     //JDA
     implementation group: 'net.dv8tion', name: 'JDA', version: '4.4.0_350'
     //Yusuf Ismail's Discord Core
-    implementation group: 'io.github.yusufsdiscordbot', name: 'application', version: '1.0.35'
+    implementation group: 'io.github.yusufsdiscordbot', name: 'jda4', version: '1.0.35'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply application/build.gradle:48]: ignore

File: application/src/main/resources/database/V2__moderation.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+(
+    case_id              BIGINT       NOT NULL PRIMARY KEY,
+    issued_at                TIMESTAMP  NOT NULL,
+    user_id                  BIGINT     NOT NULL,
+    author_id                BIGINT     NOT NULL,
+    guild_id                 BIGINT     NOT NULL,
+    action_name              TEXT       NOT NULL,
+    amount_of_times          INTEGER    NOT NULL,
+    reason                   TEXT       NOT NULL
+)
```
Comment (Zabuzard) [inline application/src/main/resources/database/V2__moderation.sql:19]: This is heavily inspired by:
https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/resources/db/V5__Add_Moderation_Actions.sql

But I do not see any attribution whatsoever, this conflicts the license.

File: application/src/main/java/io/github/yusufsdiscordbot/yusufsmoderationbot/slash_commands/moderation/ModerationHelper.java [hunk 3]
```diff
@@ -150,4 +149,29 @@ private static boolean verifyInteractionRights(YusufSlashCommandEvent yusufSlash
...
+        private final String verb;
+
+        CommandName(String verb) {
+            this.verb = verb;
+        }
+
+        public String getVerb() {
+            return verb;
+        }
+    }
```
Comment (Zabuzard) [inline application/src/main/java/io/github/yusufsdiscordbot/yusufsmoderationbot/slash_commands/moderation/ModerationHelper.java:176]: This is heavily inspired by:

https://github.com/Together-Java/TJ-Bot/blob/develop/application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java#L331

Yet, I do not see any attribution whatsoever. This conflicts the license.

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/resources/database/V2__moderation.sql (hunk 2 above)
Comment (RealYusufIsmail) [reply application/src/main/resources/database/V2__moderation.sql:19]: Will add later on

File: application/src/main/java/io/github/yusufsdiscordbot/yusufsmoderationbot/slash_commands/moderation/ModerationHelper.java (hunk 3 above)
Comment (RealYusufIsmail) [reply application/src/main/java/io/github/yusufsdiscordbot/yusufsmoderationbot/slash_commands/moderation/ModerationHelper.java:176]: Will add

**Comment by RealYusufIsmail** [comment]
Ready to be merged. can you please have a look @nDZIB

**Comment by RealYusufIsmail** [comment]
Ready to be merged. can you please have a look @nDZIB

File: application/build.gradle [hunk 4]
```diff
@@ -35,24 +36,33 @@ repositories {
 }
 
 dependencies {
+    //needed so the build gradle can find pass.
     implementation project(':database')
-
+    implementation rootProject
     //JDA
     implementation group: 'net.dv8tion', name: 'JDA', version: '4.4.0_352'
```
[the remark this answers is not in the document]
Comment (RealYusufIsmail) [reply application/build.gradle:43]: ignore

**Pull request merged**